CMakeLists.txt

```diff
@@ -81,50 +81,45 @@ endif ()
 ## Define include directories to avoid absolute paths for header defines
 include_directories(${LIBUSB_INCLUDE_DIR})
 
-# ====
-include_directories(${PROJECT_SOURCE_DIR}/include)                              ### TODO: Clean this up...
-include_directories(${PROJECT_BINARY_DIR}/include/stlink)
-include_directories(${PROJECT_SOURCE_DIR}/include/stlink)
-include_directories(${PROJECT_SOURCE_DIR}/include/stlink/tools)
-# ====
+include_directories(${PROJECT_SOURCE_DIR}/inc) # contains top-level header files
+include_directories(${PROJECT_BINARY_DIR}/inc) # contains version.h
 
 include_directories(src)
-include_directories(src/tools)                                                  ### TODO: Clean this up...
+include_directories(src/st-flash)
+include_directories(src/stlink-lib)
 
 set(STLINK_HEADERS
-        include/stlink.h
-        include/stlink/backend.h
-        include/stlink/chipid.h
-        include/stlink/commands.h
-        include/stlink/flash_loader.h
-        include/stlink/reg.h
-        src/logging.h
-        src/md5.h
-        src/sg.h
-        src/usb.h
+        inc/stlink.h
+        src/backend.h
+        src/commands.h
+        src/reg.h
+        src/stlink-lib/chipid.h
+        src/stlink-lib/flash_loader.h
+        src/stlink-lib/logging.h
+        src/stlink-lib/md5.h
+        src/stlink-lib/sg.h
+        src/stlink-lib/usb.h
         )
 
 set(STLINK_SOURCE
         src/common.c
-        src/chipid.c
-        src/flash_loader.c
-        src/logging.c
-        src/md5.c
-        src/sg.c
-        src/usb.c
+        src/stlink-lib/chipid.c
+        src/stlink-lib/flash_loader.c
+        src/stlink-lib/logging.c
+        src/stlink-lib/md5.c
+        src/stlink-lib/sg.c
+        src/stlink-lib/usb.c
         )
 
-if (WIN32 OR MINGW OR MSYS)
-    include_directories(src/mingw)
-    set(STLINK_SOURCE "${STLINK_SOURCE};src/mmap.c;src/mingw/mingw.c")
-    set(STLINK_HEADERS "${STLINK_HEADERS};src/mmap.h;src/mingw/mingw.h")
-endif ()
-
-if (MSVC)
+if (WIN32 OR MINGW OR MSYS OR MSVC)
     include_directories(src/win32)
```

Comment:
our custom unistd.h prevoiusly connected here. Only for msvc as is getopt

Reply:
I got it. Thx.